File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.kt
class Solution {
    fun solution(ingredient: IntArray): Int {
        val list = ingredient.toMutableList()
        var answer = 0
        var size = 0

        while (size < list.size-3) {
            if(list[size] == 1 && list[size+1] == 2 && list[size+2] == 3 && list[size+3] == 1) {
                repeat(4) {
                    list.removeAt(size)
                }
                if(size > 1 && list.size > 3) {
                    size -= 2   
                }else if(size > 0 && list.size > 3) {
                    size -= 1
                }
                answer++
            }else {
                size++
            }
        }
        return answer
    }
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.swift
func solution(_ ingredient:[Int]) -> Int {
    var answer = 0
    var ingredients = [Int]()
    
    for i in ingredient {
        ingredients.append(i)
        
        if ingredients.count >= 4 {
            if ingredients[ingredients.count-4...ingredients.count-1] == [1,2,3,1] {
                ingredients.removeSubrange(ingredients.count-4...ingredients.count-1)
                answer += 1
            }
        }
    }
    
    return answer
}